dynamiqs/qarrays/sparsedia_qarray.py

```diff
@@ -345,6 +345,8 @@ def _check_key_in_batch_dims(key: int | slice | tuple, ndim: int):
     valid_key = False
     if isinstance(key, (int, slice)):
         valid_key = ndim > 2
+    elif isinstance(key, Array):
+        valid_key = len(key.shape) == 0 and ndim > 2
```

Comment:
Fixed separately in https://github.com/dynamiqs/dynamiqs/pull/864 since it's unrelated to `mcsolve`.